src/api-service/__app__/onefuzzlib/workers/nodes.py

```diff
@@ -384,8 +384,9 @@ def set_shutdown(self) -> None:
 
     def set_halt(self) -> None:
         """Tell the node to stop everything."""
-        self.set_shutdown()
-        self.stop()
+        logging.info("setting halt: %s", self.machine_id)
+        self.delete_requested = True
+        self.stop(done=True)
         self.set_state(NodeState.halt)
```

Comment:
(This is not a problem with this change, but might be a separate bug?)

Reply:
It's actually on purpose.  set_shutdown specifically doesn't set state because we want the node to continue working.

Reply:
Conceptually, wouldn't `shutdown` be a state of "_scheduled_ to shutdown, but still running", while `halt` is the "don't continue working" state?

I'm actually not sure if we're using the `NodeState.shutdown` at all, right now, but that might be grep failure.

Reply:
Ah.  Moving towards that would be an impactful change across a bunch of the subsystem.  However, since we are try specifically to use NodeState to specifically mirror the node's understanding of state with a few minor exceptions, using flags to indicate what would happen once the node is done is how this was handled.

Reply:
Either way, that's a follow-on discussion